### DISCUSSION TITLE
Initial states in "PreprocessingVariationalEquation" program

### DISCUSSION THREAD

**Discussion by ReubenLi1999** [groops-devs/groops#241] [Q&A] [answered]
Hello developers,

# Description

I'm currently working with the "PreprocessingVariationalEquation" program and have stumbled upon a perplexing aspect concerning the initial state of dynamic orbits. I would greatly appreciate some elucidation from the developers to enhance my understanding of the initial state.

What I am doing now is using the "PreprocessingVariationalEquation" program to generate dynamic orbits for GRACE-FO twin satellites. My understanding is that orbit integration involves a numerical integration process with a given initial state ($r_0$ and $\dot{r}_0$ in the figure below) and a force model ($f$ in the figure below).

![image](https://github.com/groops-devs/groops/assets/49586424/17027974-1b54-4f23-9e7e-b1c4e669d0a5)

In the "PreprocessingVariationalEquation" program, I use the orbits in the inertial frame converted from GNV1B ("inputfileOrbit" in the following figure). My expectation is that the initial states of the "inputfileOrbit" and the "outputfileOrbit" are the same. However, it appears that they are different.

![image](https://github.com/groops-devs/groops/assets/49586424/0c22f5f1-449d-4750-a059-9087d97e956e)

To provide context, here's my results: The above displays the integrated dynamic orbits (output), while the below is the approximate orbits (input, converted from GNV1B to the inertial frame directly). The initial states between the two sets are different. 
![58178abf6550e5b5cd913e96ba467fd](https://github.com/groops-devs/groops/assets/49586424/93743004-303d-474b-abe1-fe377b11266e)

# Question

- Are my results reasonable? I'm concerned that I may have overlooked something.

- Have any refinements been made to the orbits after dynamic integration, similar to the approach outlined in the [Ref](https://www.sciencedirect.com/science/article/pii/S0273117717302910)?

Your time and assistance in helping me gain clarity on this matter are greatly appreciated.

Cheers,

Haosi

**Comment by tmayerguerr** [comment, marked as answer]
Hi Hasoi,

You are right, the program follows completely the approach as described in ([Ellmer and Mayer-Guerr, 2017](https://www.sciencedirect.com/science/article/pii/S0273117717302910)) with the refinement using `gradientfield` and the improved algorithm if `useEnke` is enabled. Additionally the integrated orbit is fitted to the apriori orbit by adjusting the initial state in terms of least squares adjustment.

Best regards
Torsten

**Reply by ReubenLi1999** [reply]
Thank you very much!
